### ISSUE TITLE
Middleware support

### ISSUE THREAD

**Issue by Nicolab** [EngineeringMode/Grapnel.js#18]
Hello,

it is possible with Grapnel to make a middleware? 
Like:
```js
// Middleware
router.get('product/*', function(req, event, next) {
    console.log('middleware called');
  next();
});

// Basic route
router.get('product/:id', function(req, event) {
    console.log('route called');
});
```

**Comment by supermensa** [comment]
As I understand this, then the ```next``` callback would just call the next route (if any)?

**Comment by Nicolab** [comment]
Yes like the router of Express (or [page.js](https://github.com/visionmedia/page.js)). It just pass through.

Unless I missed something, currently the only solution is to use the `match` event and check the `req.route`.

**Comment by EngineeringMode** [comment]
Grapnel automatically calls the equivalent of `next()` unless `event.stopPropagation()` is called somewhere in the stack. However, the `req` object is different for route handlers that are called later on. So the concept of middleware is already there.

The solution is simple: allow `req` to be accessed by other handlers in the stack.

```javascript

router.get('/profile/*', function(req, e){
    if(user.auth()){
        req.user = user;
    }else{
        e.stopPropagation();
        // Handle
    }
});

router.get('/profile/:user', function(req, e){
    console.log(req.user.id);
});
```

I'll add this to the next release.

**Comment by Nicolab** [comment]
Excellent, thanks.
Yes, the first solution is more simple and quick to write.

**Comment by Nicolab** [comment]
A suggestion, may be coupled with #19 
It would be great!

**Comment by EngineeringMode** [comment]
@Nicolab @supermensa 
Landed in `v0.5.3`

**Comment by Nicolab** [comment]
It's excellent, thanks :)
